pkg/graph/resolvers/nda_agreement.go

```diff
@@ -0,0 +1,77 @@
+package resolvers
+
+import (
+	"time"
+
+	"github.com/google/uuid"
+	"go.uber.org/zap"
+
+	"github.com/cmsgov/mint-app/pkg/authentication"
+	"github.com/cmsgov/mint-app/pkg/graph/model"
+	"github.com/cmsgov/mint-app/pkg/models"
+	"github.com/cmsgov/mint-app/pkg/storage"
+)
+
+//NDAAgreementGetByEUA returns an EUA agreement by eua
+func NDAAgreementGetByEUA(logger *zap.Logger, principal authentication.Principal, store *storage.Store) (*model.NDAInfo, error) {
+	nda, err := store.NDAAgreementGetByEUA(logger, principal.ID())
+	accepted := model.NDAInfo{}
+	if err != nil {
+		return nil, err
+	}
+	if nda == nil {
```

Comment:
Is it worth defensive programming here, and doing a check to see if `agreed` is `true`, but `agreedDts` is `NULL`, and throwing an error?

I can't really imagine a case (right now) that would cause this, but always try and think about cases like this!

Reply:
(not requesting changes here, specifically, this one's more food for thought)